### PR TITLE
[css-grid] Fix bug when grid-area rules are not inserted

### LINKED ISSUE #1191: [css-grid] media rules are not inserted for grid-template[-areas] inside media queries
**Description**
There is a new bug where there is no `lastRule` set when `grid-template-areas` are declared within a media query. The `insertAreas()` utility function attempts to media rules for each area, but fails because it has no designated rule to append them to.
*NOTE:* This was working correctly in version 9.1.5.

**Expected Result**
For `grid-template-areas` declared within media queries, new media queries are inserted for each `grid-area` defined in the template.

**Proposed Solution**
I believe the bug was introduced in version 9.2:
https://github.com/postcss/autoprefixer/blob/2072e73d77fd1abc67e49f74950e3df2d6599026/lib/hacks/grid-utils.js#L606-L608
The issue goes away if the `if` condition is removed and you, instead, just have:
```
        rulesToInsert[lastArea].lastRule = gridAreaMedia || gridAreaRule
```

**CSS Example to Reproduce**
Input:
```
.example {
  display: grid;
}

@media (max-width: 767.98px) {
  .example {
    grid-template-areas:
      'first first'
      'second third';
  }
}

.first {
  grid-area: first;
}
  
.second {
  grid-area: second;
}

.third {
  grid-area: third;
}
```

Current output:
```
.example {
  display: -ms-grid;
  display: grid;
}

@media (max-width: 767.98px) {
  .example {
        grid-template-areas:
      'first first'
      'second third';
  }
}

.first {
  grid-area: first;
}
  
.second {
  grid-area: second;
}

.third {
  grid-area: third;
}
```

Expected output:
```
.example[_ngcontent-c23] {
  display: -ms-grid;
  display: grid;
}

@media (max-width: 767.98px) {
  .example[_ngcontent-c23] {
        grid-template-areas: 'first first' 'second third';
  }
}

.first[_ngcontent-c23] {
  grid-area: first;
}

.second[_ngcontent-c23] {
  grid-area: second;
}

.third[_ngcontent-c23] {
  grid-area: third;
}

@media (max-width: 767.98px) {
  .first[_ngcontent-c23] {
    -ms-grid-row: 1;
    -ms-grid-column: 1;
    -ms-grid-column-span: 2;
  }
  .second[_ngcontent-c23] {
    -ms-grid-row: 2;
    -ms-grid-column: 1;
  }
  .third[_ngcontent-c23] {
    -ms-grid-row: 2;
    -ms-grid-column: 2;
  }
}
```

*NOTE:* While I was investigating this, I also noticed these lines:
https://github.com/postcss/autoprefixer/blob/2072e73d77fd1abc67e49f74950e3df2d6599026/lib/hacks/grid-utils.js#L534-L537
I have not run into any issues from this, but it looks like it was meant to be something like:
```
        if (rulesToInsert[lastArea].lastRule) {
          rulesToInsert[lastArea].lastRule.after(cloned)
        } else {
          rulesToInsert[lastArea].lastRule = cloned
        }
```

### PR DESCRIPTION
**Pull request by bogdan0083**
Fixes #1191 

/cc @Dan503 for PR review 🙂

@cekejiuba We would be happy if you could test the fix in your spare time. Here are the things you need to do:

1. Clone my branch: `git clone -b fix/grid-areas-bug https://github.com/bogdan0083/autoprefixer`
2. run `yarn` inside cloned repository
3. run `npx gulp play`, add the code to `./playground/input.css` and save the file
4. the output will be in `./playground/output.css`

### CODE REVIEW

File: test/cases/grid-template-areas.css [hunk 1]
```diff
@@ -215,3 +215,27 @@
...
+  grid-area: first;
+}
+  
+.grid-area-second {
+  grid-area: second;
+}
+
+.grid-area-third {
+  grid-area: third;
+}
```
Comment (ai) [inline test/cases/grid-template-areas.css]: Can you insert new line in both CSS files?

**Comment by cekejiuba** [comment]
@bogdan0083 I tested the fix and verify it fixes https://github.com/postcss/autoprefixer/issues/1191

**Comment by Dan503** [comment]
Sorry, I'm sick at the moment and not really in a state to be doing any testing. 🤒

I'm wondering how this issue even came to exist though. Don't we already have tests in place that tested for this? I'm wondering how this issue ever got into production. 😕

**Comment by cekejiuba** [comment]
From my understanding, this appears to be an edge case where:
1. No `grid-area` declarations are nested in a media query.
2. All `grid-template-area` (or, likely, `grid-template`) declarations are defined inside media queries.
3. All `grid-template-area` declarations include each of the available grid areas.

If (2) was not true, there were other `if` blocks that would set the `rulesToInsert[lastArea].lastRule` unconditionally in their last line.
https://github.com/postcss/autoprefixer/blob/master/lib/hacks/grid-utils.js#L526-L539
https://github.com/postcss/autoprefixer/blob/master/lib/hacks/grid-utils.js#L539-L561

If (3) was not true, there was a separate `if` block that would set the `rulesToInsert[lastArea].lastRule` unconditionally.
https://github.com/postcss/autoprefixer/blob/master/lib/hacks/grid-utils.js#L511-L518

In other words, the issue specifically appeared when the `grid-area` declarations were not defined inside media queries, but all `grid-template` or `grid-template-areas` declarations that referenced them _were_ inside media queries.

Essentially, this code will fail in any case where there are `rulesToInsert`, but `lastRule` is undefined:
https://github.com/postcss/autoprefixer/blob/master/lib/hacks/grid-utils.js#L621-L634

File: test/cases/grid-template-areas.css (hunk 1 above)
Comment (bogdan0083) [reply test/cases/grid-template-areas.css]: Done 👍

**Comment by bogdan0083** [comment]
> Sorry, I'm sick at the moment and not really in a state to be doing any testing. 🤒

Get well Dan! ❤️

> I'm wondering how this issue even came to exist though. Don't we already have tests in place that tested for this? I'm wondering how this issue ever got into production. 😕

@cekejiuba described the edge case perfectly. 

The thing is that the algorithm is quite tricky but the code I wrote is quite dirty (although I tried my best at that time). I need to rewrite that part of the codebase someday to not feel bad about it 🤔.

**Comment by ai** [comment]
Are we ready to release it?

Did I understand correctly, that @cekejiuba found a few edge cases not covered by this PR?

If this solution is not perfect, but it closes a few issues without opening new issues, I think it is still a good deal for release.

**Comment by bogdan0083** [comment]
> Are we ready to release it?

I guess we are. The PR fixes the original issue.

> Did I understand correctly, that @cekejiuba found a few edge cases not covered by this PR?

Nope. @cekejiuba described only one edge case and provided steps how to reproduce it. This PR fixes that case:

> In other words, the issue specifically appeared when the `grid-area` declarations were not defined inside media queries, but all `grid-template` or `grid-template-areas` declarations that referenced them _were_ inside media queries.
I presume this specific scenario was not covered in the tests.

**Comment by Dan503** [comment]
Does this PR also test for that edge case?

**Comment by bogdan0083** [comment]
@Dan503 Yep. Here's the tests I added to make sure its fixed: https://github.com/postcss/autoprefixer/commit/dc296474c9882e482c43222b5301185125a54b46

**Comment by ai** [comment]
Released in 9.4.9
